File: FarmaciaERP/Backend/src/main/java/FarmaciaERP/Application/UseCases/RevisarExcepcionFacturacionUseCase.java
package FarmaciaERP.Application.UseCases;

import FarmaciaERP.Application.DTOs.Response.ExcepcionFacturacionResponse;
import FarmaciaERP.Domain.Entities.ExcepcionFacturacion;
import FarmaciaERP.Domain.Exceptions.BadRequestException;
import FarmaciaERP.Domain.Repositories.IExcepcionFacturacionRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/**
 * FI-AP.01 Fase 01 - paso 1.1: el Analista de Cuentas por Pagar revisa el
 * panel de facturas bloqueadas.
 */
@Service
public class RevisarExcepcionFacturacionUseCase {

    private final IExcepcionFacturacionRepository excepcionFacturacionRepository;

    public RevisarExcepcionFacturacionUseCase(IExcepcionFacturacionRepository excepcionFacturacionRepository) {
        this.excepcionFacturacionRepository = excepcionFacturacionRepository;
    }

    @Transactional
    public ExcepcionFacturacionResponse revisar(Long id) {
        ExcepcionFacturacion excepcion = excepcionFacturacionRepository.findById(id)
                .orElseThrow(() -> new BadRequestException("Excepción de facturación no encontrada: " + id));

        excepcion.revisar();

        ExcepcionFacturacion guardada = excepcionFacturacionRepository.save(excepcion);
        return ExcepcionFacturacionResponseAssembler.toResponse(guardada);
    }
}

File: FarmaciaERP/frontend-react/src/domain/usecases/excepcionFacturacion/revisarExcepcionFacturacionUseCase.js
import { ExcepcionFacturacion } from '../../models/ExcepcionFacturacion';

// Analista AP · paso 1.1: Revisar Panel de Facturas Bloqueadas.
export const revisarExcepcionFacturacionUseCase = (excepcionFacturacionRepository) => ({
  async execute(id) {
    if (!id) throw new Error('El id de la excepción de facturación es requerido');

    const raw = await excepcionFacturacionRepository.revisar(id);
    return ExcepcionFacturacion.fromApi(raw);
  },
});
